WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductRowViewModelTests.swift

```diff
@@ -426,10 +426,193 @@ final class ProductRowViewModelTests: XCTestCase {
+        XCTAssertEqual(viewModel.name, name)
+        XCTAssertNotNil(viewModel.productSubscriptionDetails)
+    }
+
+    func test_productRow_when_product_type_is_not_subscription_but_contains_subscription_metadata_then_productRow_has_no_subscription_metadata() {
+        // Given
+        let rowID = Int64(0)
+        let subs: ProductSubscription = createFakeSubscription()
+        let productTypeKey = "simple"
+
+        let product = Product.fake().copy(productID: 12,
+                                          name: "Not a subscription product anymore, but might have subscription metadata",
+                                          productTypeKey: productTypeKey,
+                                          subscription: subs)
+
+        // When
+        let viewModel = ProductRowViewModel(id: rowID, product: product, productSubscriptionDetails: subs)
+
+        // Then
+        XCTAssertEqual(viewModel.id, rowID)
+        XCTAssertEqual(viewModel.productOrVariationID, product.productID)
+        XCTAssertEqual(viewModel.name, product.name)
+        XCTAssertNil(viewModel.productSubscriptionDetails)
+    }
+
+    func test_subscriptionBillingDetailsLabel_when_periodInterval_is_1_then_returns_singular_details() {
+        // Given
+        let rowID = Int64(0)
+        let subs: ProductSubscription = createFakeSubscription()
+        let productTypeKey = "subscription"
+
+        let product = Product.fake().copy(productID: 12,
+                                          name: "A subscription product",
+                                          productTypeKey: productTypeKey,
+                                          subscription: subs)
+
+        // When
+        let viewModel = ProductRowViewModel(id: rowID, product: product, productSubscriptionDetails: subs)
+
+        // Then
+        XCTAssertEqual(viewModel.subscriptionBillingDetailsLabel, "$5.00 / 1 month")
+    }
+
+    func test_subscriptionBillingDetailsLabel_when_periodInterval_is_more_than_1_then_returns_pluralized_details() {
+        // Given
+        let rowID = Int64(0)
+        let subs: ProductSubscription = createFakeSubscription(periodInterval: "3")
+        let productTypeKey = "subscription"
+
+        let product = Product.fake().copy(productID: 12,
+                                          name: "A subscription product",
+                                          productTypeKey: productTypeKey,
+                                          subscription: subs)
+
+        // When
+        let viewModel = ProductRowViewModel(id: rowID, product: product, productSubscriptionDetails: subs)
+
+        // Then
+        XCTAssertEqual(viewModel.subscriptionBillingDetailsLabel, "$5.00 / 3 months")
+    }
+
+    func test_subscriptionConditionsLabel_when_has_signup_fees_and_trial_period_then_returns_expected_details() {
+        // Given
+        let rowID = Int64(0)
+        let subs: ProductSubscription = createFakeSubscription(periodInterval: "3")
```

Comment:
Nit: we should set the values explicitly here for clarity even though there are default values in the helper method. That'll make it easier to understand what the conditions are.

Reply:
Yep! Updated tests on 48f82173d4a186055f9a05467d3ec6b13a630ff9